From f34d32c8bb2325e9923f6fa3d3ef1bcd1e238cce Mon Sep 17 00:00:00 2001
From: Frieder Schrempf <frieder.schrempf@kontron.de>
Date: Thu, 12 Jan 2023 18:54:43 +0100
Subject: [PATCH] imx-base.inc: Make UBOOT_ENTRYPOINT overwriteable

This is useful if for some reason other values need to be defined in
overlayed layers. For example to fix the kernel alignment warning
since kernel 5.8 by setting:

UBOOT_ENTRYPOINT:mx8m-generic-bsp = "0x40400000"

Signed-off-by: Frieder Schrempf <frieder.schrempf@kontron.de>
---
 conf/machine/include/imx-base.inc | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/conf/machine/include/imx-base.inc b/conf/machine/include/imx-base.inc
index 04ad558d..d02ca76c 100644
--- a/conf/machine/include/imx-base.inc
+++ b/conf/machine/include/imx-base.inc
@@ -83,20 +83,20 @@ PREFERRED_PROVIDER_virtual/bootloader ??= "${IMX_DEFAULT_BOOTLOADER}"
 
 PREFERRED_PROVIDER_u-boot-mxsboot-native ??= "u-boot-fslc-mxsboot-native"
 
-UBOOT_ENTRYPOINT:mxs-generic-bsp    = "0x40008000"
-UBOOT_ENTRYPOINT:mx51-generic-bsp   = "0x90008000"
-UBOOT_ENTRYPOINT:mx53-generic-bsp   = "0x70008000"
-UBOOT_ENTRYPOINT:mx6-generic-bsp    = "0x10008000"
-UBOOT_ENTRYPOINT:mx6sl-generic-bsp  = "0x80008000"
-UBOOT_ENTRYPOINT:mx6sll-generic-bsp = "0x80008000"
-UBOOT_ENTRYPOINT:mx6sx-generic-bsp  = "0x80008000"
-UBOOT_ENTRYPOINT:mx6ul-generic-bsp  = "0x80008000"
-UBOOT_ENTRYPOINT:mx6ull-generic-bsp = "0x80008000"
-UBOOT_ENTRYPOINT:mx6ulz-generic-bsp = "0x80008000"
-UBOOT_ENTRYPOINT:mx7-generic-bsp    = "0x80008000"
-UBOOT_ENTRYPOINT:mx7ulp-generic-bsp = "0x60008000"
-UBOOT_ENTRYPOINT:mx8m-generic-bsp   = "0x40480000"
-UBOOT_ENTRYPOINT:vf-generic-bsp     = "0x80008000"
+UBOOT_ENTRYPOINT:mxs-generic-bsp    ?= "0x40008000"
+UBOOT_ENTRYPOINT:mx51-generic-bsp   ?= "0x90008000"
+UBOOT_ENTRYPOINT:mx53-generic-bsp   ?= "0x70008000"
+UBOOT_ENTRYPOINT:mx6-generic-bsp    ?= "0x10008000"
+UBOOT_ENTRYPOINT:mx6sl-generic-bsp  ?= "0x80008000"
+UBOOT_ENTRYPOINT:mx6sll-generic-bsp ?= "0x80008000"
+UBOOT_ENTRYPOINT:mx6sx-generic-bsp  ?= "0x80008000"
+UBOOT_ENTRYPOINT:mx6ul-generic-bsp  ?= "0x80008000"
+UBOOT_ENTRYPOINT:mx6ull-generic-bsp ?= "0x80008000"
+UBOOT_ENTRYPOINT:mx6ulz-generic-bsp ?= "0x80008000"
+UBOOT_ENTRYPOINT:mx7-generic-bsp    ?= "0x80008000"
+UBOOT_ENTRYPOINT:mx7ulp-generic-bsp ?= "0x60008000"
+UBOOT_ENTRYPOINT:mx8m-generic-bsp   ?= "0x40480000"
+UBOOT_ENTRYPOINT:vf-generic-bsp     ?= "0x80008000"
 
 # Some derivates can utilize the boot container provided by U-Boot,
 # below variable sets that those machines which have a imx-boot-container
-- 
2.39.0

